### PR TITLE
Updated with php 8.4

### PR DESCRIPTION
**Pull request by SuperDJ**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

- Added PHP 8.4;
- Updated `squizlabs/php_codesniffer` to use the [new repository](https://github.com/PHPCSStandards/PHP_CodeSniffer/);

### CODE REVIEW

**Review by gsteel — CHANGES_REQUESTED** [review]
There are PHP deprecations getting emitted during tests:

https://github.com/laminas/laminas-xml/actions/runs/11254698819/job/31292696978?pr=19#step:3:380

These will need to be fixed _(and PHPUnit should be configured to `failOnDeprecation`)_

There are 4 implicit null arguments by the look of it

**Comment by SuperDJ** [comment]
@gsteel Unfortunately the optional parameter order is deprecated as well https://github.com/laminas/laminas-xml/actions/runs/11274526733/job/31353887275?pr=19#step:3:231

File: src/Security.php [hunk 1]
```diff
@@ -28,13 +28,13 @@ protected static function heuristicScan($xml)
...
      * @param   string $xml
-     * @param   DomDocument $dom
+     * @param   DomDocument|null $dom
      * @param   int $libXmlConstants additional libxml constants to pass in
      * @param   callable $callback the callback to use to create the dom element
      * @throws  Exception\RuntimeException
      * @return  SimpleXMLElement|DomDocument|boolean
      */
-    private static function scanString($xml, DOMDocument $dom = null, $libXmlConstants, callable $callback)
+    private static function scanString($xml, DOMDocument|null $dom = null, $libXmlConstants, callable $callback)
```
Comment (gsteel) [inline src/Security.php]:
```suggestion
    private static function scanString($xml, DOMDocument|null $dom, $libXmlConstants, callable $callback)
```

This is a private method - you can do anything you like to it!

File: phpunit.xml.dist [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
-            <directory suffix=".php">./src</directory>
-        </include>
-    </coverage>
-
+<phpunit
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
+    bootstrap="vendor/autoload.php"
+    colors="true"
+    failOnDeprecation="true"
```
Comment (gsteel) [inline phpunit.xml.dist:15]: At the risk of being a pain in the ass, would you mind adding these too?:

- `failOnWarning`
- `failOnNotice`

**Review by gsteel — CHANGES_REQUESTED** [review]

File: phpunit.xml.dist (hunk 2 above)
Comment (gsteel) [reply phpunit.xml.dist:15]: Also, these are helpful:

```xml
        displayDetailsOnTestsThatTriggerDeprecations="true"
        displayDetailsOnTestsThatTriggerErrors="true"
        displayDetailsOnTestsThatTriggerNotices="true"
        displayDetailsOnTestsThatTriggerWarnings="true"
```

File: composer.json [hunk 3]
```diff
@@ -22,15 +22,15 @@
...
         "ext-simplexml": "*"
     },
     "require-dev": {
-        "laminas/laminas-coding-standard": "~1.0.0",
-        "squizlabs/php_codesniffer": "3.6.1 as 2.9999999.9999999",
-        "phpunit/phpunit": "^9.5.8",
-        "ext-iconv": "*"
+		"ext-iconv": "*",
+		"laminas/laminas-coding-standard": "~1.0.0",
+		"phpunit/phpunit": "^10.5 || ^11.4",
```
Comment (gsteel) [inline composer.json]: Bumping min version of phpunit to 10.5.35 should fix CI failures on 8.4 lowest

**Review by gsteel — CHANGES_REQUESTED** [review]
@SuperDJ If you could trim trailing whitespace in the readme and fix the min PHPunit version, we should have green CI

Thanks for sticking with it!

**Review by gsteel — APPROVED** [review]
Thanks @SuperDJ - Much appreciated :)

**Pull request merged**